src/components/widgets/Map.vue

```diff
@@ -265,13 +323,78 @@ onBeforeMount(() => {
 
 onBeforeUnmount(() => {
   clearInterval(followInterval)
+  if (leafletMap) {
+    leafletMap.off('click', onMapClick)
+  }
+})
+
+onBeforeUnmount(() => {
+  window.removeEventListener('keydown', onKeydown)
+  if (leafletMap) {
+    leafletMap.off('contextmenu')
+  }
 })
+
+const showContextMenu = ref(false)
+const menuPosition = reactive({ top: '0px', left: '0px' })
+
+// eslint-disable-next-line no-undef
+const onMapClick = (event: L.LeafletMouseEvent): void => {
+  console.log('Map click event:', event) // Log the event object
+
+  // Check if event.latlng is defined and has the required properties
+  if (event?.latlng?.lat != null && event?.latlng?.lng != null) {
+    clickedLocation.value = [event.latlng.lat, event.latlng.lng]
+    showContextMenu.value = true
+
+    // Calculate and update menu position
+    const mapElement = leafletObject.value?.getContainer()
+    if (mapElement) {
+      const { x, y } = mapElement.getBoundingClientRect()
+      menuPosition.left = `${event.originalEvent.clientX - x}px`
+      menuPosition.top = `${event.originalEvent.clientY - y}px`
+    }
+  } else {
+    console.error('Invalid event structure:', event)
+  }
+}
+
+const onMenuOptionSelect = (option: string): void => {
+  console.log(`Option selected: ${option}`)
+
+  switch (option) {
+    case 'goto':
+      if (clickedLocation.value) {
+        // Define default values
+        const hold = 0
+        const acceptanceRadius = 0
+        const passRadius = 0
+        const yaw = 0
+        const altitude = vehicleStore.coordinates.altitude ?? 0
+
+        const latitude = clickedLocation.value[0]
+        const longitude = clickedLocation.value[1]
+
+        vehicleStore.goto(hold, acceptanceRadius, passRadius, yaw, latitude, longitude, altitude)
+      }
+      break
+
+    // Add more cases for other options if needed in the future
+
+    default:
+      console.warn('Unknown menu option selected:', option)
+  }
+
+  // hide the context menu after an option is selected
+  showContextMenu.value = false
+}
```

Comment:
This context menu on right click is not appearing for me.

Reply:
I was going for left click show menu and if you want to cancel the command and make the menu go a away you can either press escape or right click

Reply:
Right! If you prefer to leave this for a separate PR its also fine, as the goTo works without it.

Reply:
I think its working as I intended, no?